File: js/9935.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

let s = inputs[0];
const bombS = inputs[1];
const stack = [];

for (let i = 0; i < s.length; i++) {
  if (s[i] === bombS[bombS.length - 1]) {
    let flag = false;
    for (let j = 0; j < bombS.length - 1; j++) {
      if (stack[stack.length - bombS.length + 1 + j] !== bombS[j]) {
        flag = true;
        break;
      }
    }

    if (flag) {
      stack.push(s[i]);
    } else {
      for (let j = 0; j < bombS.length - 1; j++) {
        stack.pop();
      }
    }
  } else {
    stack.push(s[i]);
  }
}

console.log(stack.length === 0 ? 'FRULA' : stack.join(''));

File: c++/9935.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<char> stack;

int main() {
  freopen("example.txt", "r", stdin);

  string s, bombS;
  cin >> s >> bombS;
  int bombSize = bombS.length();

  for (int i = 0; i < s.length(); i++) {
    if (s[i] == bombS[bombSize - 1]) {
      bool flag = false;

      if (stack.size() >= bombSize - 1) {
        for (int j = 0; j < bombSize - 1; j++) {
          if (stack[stack.size() - bombSize + 1 + j] != bombS[j]) {
            flag = true;
            break;
          }
        }
      } else {
        flag = true;
      }

      if (flag)
        stack.push_back(s[i]);
      else {
        for (int j = 0; j < bombSize - 1; j++) {
          stack.pop_back();
        }
      }
    } else {
      stack.push_back(s[i]);
    }
  }

  if (stack.size() == 0) {
    cout << "FRULA";
  } else {
    for (int i = 0; i < stack.size(); i++) {
      cout << stack[i];
    }
  }

  return 0;
}